### DISCUSSION TITLE
How to create multiple VU and make them collaborate?

### DISCUSSION THREAD

**Discussion by AdrKacz** [artilleryio/artillery#1421] [Artillery]
# How to create multiple VU and make them collaborate?

Hi 👋 

I am working on a project that involves collaboration between users. One user can send a request to the server. This server will redirect the request to a sub-set of other users, who will have to answer to determine the final behaviour. All of this happens via **WebSocket**.

Thanks to *Artillery* I can create numerous VU. However, I didn't find a way to build a *small box of VUs* where I could send messages according to what VU received from other VUs. Ideally, I could create many of these * small boxes of VU* to test the load when many groups of users collaborate on different problems.

If that's an easy problem with the solution on the documentation, please redirect me 📚

**Comment by hassy** [comment]
hi @AdrKacz! 👋 that's an interesting use-case. We'd need to have a way to subscribe VUs to incoming messages + fire off messages in response based on some logic for this to work I think? How complex would that logic be?

**Reply by AdrKacz** [reply]
I have no idea at all. I quickly look through the code and find a quick way to do a *similar* job (#917).

That quick fix is suitable for waiting for an answer from a server and could be used to simulate a conversation if you know in advance what message you should receive. But I'm very sure there is a better solution.
